azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -268,20 +268,19 @@ private void updateExecutableFlow(final ExecutableFlow flow, final EncodingType
       }
     } catch (final IOException e) {
       flow.setStatus(Status.FAILED);
-      updateExecutableFlowStatus(flow);
+      updateExecutableFlowStatusInDB(flow);
       throw new ExecutorManagerException("Error encoding the execution flow. Execution Id  = "
           + flow.getExecutionId());
     } catch (final RuntimeException re) {
       flow.setStatus(Status.FAILED);
       // Likely due to serialization error
       if ( data == null && re instanceof NullPointerException) {
```

Comment:
Are we expecting NullPointerException here? Looks like it is handled in this PR and also in the frontend code.

Reply:
We are expecting that if there is an exception we should be able to handle it.
Since NPE is most likely to happen, we want to log the stack trace immediately for better debuggability.